paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java

```diff
@@ -783,6 +783,10 @@ private void generateNestedColumnUpdates(
         }
 
         if (oldType.isNullable() != newType.isNullable()) {
+            Preconditions.checkArgument(
```

Comment:
We should modify `SchemaManager.generateTableSchema` instead of here.

Reply:
This is done.